### DISCUSSION TITLE
Display of green energy in Navigator Pro

### DISCUSSION THREAD

**Discussion by KVS62** [kodebach/hacs-idm-heatpump#138] [Q&A] [answered]
I am using the integration as well as a script to provide information about solar surplus as well as solar generation to the heat pump. This is used to optimize the run schedule when there is surplus of solar energy and this works well.

In some of the illustrations listed in IDM's Navigator Pro documentation, there are screenprints of statistics called Stroomfluss. I do not see these in my Navigator App. Does anybody know whether this could be because 1) I am not transferring all the required data to the heat pump (e.g. house consumption) or 2) it is something that first needs to be activated by my installer?

**Comment by kodebach** [comment]
In the settings of the IDM Navigator Pro interface do you see an option "Photovoltaik"?

If not this probably needs to be enabled (hidden behind "Fachmann Codeeingabe"). AFAIK there are different options for the PV signal. The one for using Modbus TCP (e.g. via the integration) is probably called "Gebäudeleittechnik"/"GLT" or similar. If in doubt call IDM support, they are normally quite helpful in such cases.

**Reply by KVS62** [reply]
Thanks for answering. Photovoltaik is activated; I can see the surplus solar energy in the Navigator App and the optimization works well. I will check this with IDM support unless anyone else has some good advice for how to get the data displayed.

**Comment by KVS62** [comment, marked as answer]
In order to get the extended graphic overview in the IDM App for solar energy use it is required to write:

- Solar surplus to register 74 - Use both the consumption/feed-in value for the house connection - value should be positive for feed-in or negative for consumption
- PV production to register 78 - Always positive
- House consumption to register 82 - Is the sum of consumption/feed-in for the house connection + the PV production - important difference from 74: Consumption is positive and surplus is negative for the house connection 

You will then get the more extensive graphic in the App.

![IMG_9870](https://github.com/user-attachments/assets/6ebb7204-b2bb-4637-b2ab-3471b26bb542)

**Reply by kodebach** [reply]
There is also 84 and 86 for battery info. Sorry, I couldn't be of more help. Should've mentioned that my Fronius solar inverter connected all this automatically.

See also [the PDF I just found](https://www.idm-energie.at/wp-content/uploads/2021/04/building-management-system.pdf) while searching again for info about the battery part.

**Reply by aivatco** [reply]
hey,  
a quetion, is it possible that this funktions are not in older software´s. i have always errors.
Failed to fetch registers for group 82 (count=5): ExceptionResponse(dev_id=1, function_code=132, exception_code=2)
Failed to fetch registers for group 1748 (count=16): ExceptionResponse(dev_id=1, function_code=132, exception_code=2)

the heatpump is on version  20.16-1. 

I also cannot write to the household consumption register.

the trace from my automation
__
Fehler: AssertionError
Ergebnis:
params:
  domain: idm_heatpump
  service: set_power
  service_data:
    target: sensor.idm_warmepumpe_hausverbrauch
    value: 0.71
    acknowledge_risk: true
  target: {}
running_script: false
__
the surplus works. oh i hate modbus.

thanks for your answers.
